sdccc/src/test/java/com/draeger/medical/sdccc/tests/glue/invariant/InvariantRemoteControlCapabilitiesTestTest.java

```diff
@@ -0,0 +1,260 @@
+    @Test
+    public void testRequirementR00780Good() throws Exception {
+        final var instanceIdentifier = participantFactory.createInstanceIdentifier();
+        instanceIdentifier.setRootName(EXPECTED_ROOT);
+        instanceIdentifier.setExtensionName(EXPECTED_EXTENSION);
+        final var part = buildOperationInvokedReportReportPart(
+                "handle", messageBuilder.buildInvocationInfo(1L, InvocationState.START), instanceIdentifier);
+        final var report = buildOperationInvokedReport(MdibBuilder.DEFAULT_SEQUENCE_ID, BigInteger.ONE, part);
+        messageStorageUtil.addInboundSecureHttpMessage(storage, report);
+        testClass.testRequirementR00780();
+    }
+
+    /**
+     * Tests if the test case fails when a report part does not have the expected root in their invocation source set.
+     *
+     * @throws Exception on any exception
+     */
+    @Test
+    public void testRequirementR00780BadRootName() throws Exception {
+        final var notTheRoot = "not the expected root name";
+        final var instanceIdentifier = participantFactory.createInstanceIdentifier();
+        instanceIdentifier.setRootName(notTheRoot);
+        instanceIdentifier.setExtensionName(EXPECTED_EXTENSION);
+        final var part = buildOperationInvokedReportReportPart(
+                "handle", messageBuilder.buildInvocationInfo(1L, InvocationState.START), instanceIdentifier);
+        final var report = buildOperationInvokedReport(MdibBuilder.DEFAULT_SEQUENCE_ID, BigInteger.ONE, part);
+        messageStorageUtil.addInboundSecureHttpMessage(storage, report);
+        final var error = assertThrows(AssertionError.class, testClass::testRequirementR00780);
+        assertTrue(error.getMessage().contains(notTheRoot));
+    }
+
+    /**
+     * Tests if the test case fails when a report part does not have the expected extension in their invocation source
+     * set.
+     *
+     * @throws Exception on any exception
+     */
+    @Test
+    public void testRequirementR00780BadExtensionName() throws Exception {
+        final var notTheExtension = "not the expected extension name";
+        final var instanceIdentifier = participantFactory.createInstanceIdentifier();
+        instanceIdentifier.setRootName(EXPECTED_ROOT);
+        instanceIdentifier.setExtensionName(notTheExtension);
+        final var part = buildOperationInvokedReportReportPart(
+                "handle", messageBuilder.buildInvocationInfo(1L, InvocationState.START), instanceIdentifier);
+        final var report = buildOperationInvokedReport(MdibBuilder.DEFAULT_SEQUENCE_ID, BigInteger.ONE, part);
+        messageStorageUtil.addInboundSecureHttpMessage(storage, report);
+        final var error = assertThrows(AssertionError.class, testClass::testRequirementR00780);
+        assertTrue(error.getMessage().contains(notTheExtension));
+    }
+
+    /**
+     * Tests if the test case fails when a report part does not have the expected extension or root in their invocation
```

Comment:
Tests if the test case fails when a report part does not have the expected root in their...

This unit test does not test that it also fails in the case of an unexpected extension, hence the "or"-part is misleading.

Reply:
In order to test that as well, you would need a second unit test.

Reply:
changed the unit test description